secret/secret_scanner_test.go

```diff
@@ -0,0 +1,251 @@
+			name: "rule exclude block",
+			fields: fields{
+				Rules: []secret.Rule{
+					{
+						ID:       basicRule.ID,
+						Type:     basicRule.Type,
+						Severity: basicRule.Severity,
+						Title:    basicRule.Title,
+						Regex:    basicRule.Regex,
+						ExcludeBlocks: secret.ExcludeBlocks{
+							Title: "Exclude blocks",
+							Regexes: []*regexp.Regexp{
+								regexp.MustCompile(`--- ignore block start ---(.|\s)*--- ignore block stop ---`),
+							},
+						},
+					},
+				},
+			},
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{
+					{
+						RuleID:    basicRule.ID,
+						Type:      basicRule.Type,
+						Severity:  basicRule.Severity,
+						Title:     basicRule.Title,
+						StartLine: 4,
+						EndLine:   5,
+						Match:     "*****",
+					},
+				},
+			},
+		},
+		{
+			name: "global allowlist path",
+			fields: fields{
+				Rules: []secret.Rule{
+					basicRule,
+				},
+				AllowList: secret.AllowList{
+					Title: "Allowlist",
+					Paths: []*regexp.Regexp{
+						regexp.MustCompile(`test`),
+					},
+				},
+			},
+			inputFilePath: "testdata/secret.txt",
+			want:          types.Secret{},
+		},
+		{
+			name: "global allowlist regex",
+			fields: fields{
+				Rules: []secret.Rule{
+					basicRule,
+				},
+				AllowList: secret.AllowList{
+					Title: "Allowlist",
+					Regexes: []*regexp.Regexp{
+						regexp.MustCompile(`some`),
+					},
+				},
+			},
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{
+					{
+						RuleID:    basicRule.ID,
+						Type:      basicRule.Type,
+						Severity:  basicRule.Severity,
+						Title:     basicRule.Title,
+						StartLine: 4,
+						EndLine:   5,
+						Match:     "*****",
+					},
+				},
+			},
+		},
+		{
+			name: "global exclude block",
+			fields: fields{
+				Rules: []secret.Rule{
+					{
+						ID:       basicRule.ID,
+						Type:     basicRule.Type,
+						Severity: basicRule.Severity,
+						Title:    basicRule.Title,
+						Regex:    basicRule.Regex,
+					},
+				},
+				ExcludeBlocks: secret.ExcludeBlocks{
+					Title: "Exclude blocks",
+					Regexes: []*regexp.Regexp{
+						regexp.MustCompile(`--- ignore block start ---(.|\s)*--- ignore block stop ---`),
+					},
+				},
+			},
+			inputFilePath: "testdata/secret.txt",
+			want: types.Secret{
+				FilePath: "testdata/secret.txt",
+				Findings: []types.SecretFinding{
+					{
+						RuleID:    basicRule.ID,
+						Type:      basicRule.Type,
+						Severity:  basicRule.Severity,
+						Title:     basicRule.Title,
+						StartLine: 4,
+						EndLine:   5,
+						Match:     "*****",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := secret.NewScanner("", tt.fields.Rules, tt.fields.AllowList, tt.fields.ExcludeBlocks)
+			content, err := ioutil.ReadFile(tt.inputFilePath)
```

Comment:
ioutil is deprecated.
```suggestion
			content, err := os.ReadFile(tt.inputFilePath)
```

Reply:
fixed